### PR TITLE
ignore non-Java files in enum static init unwind

### PR DESCRIPTION
**Pull request by mgudemann**
To correctly unwind static inits of Java enums, one first has to check whether
the function is actually a Java function. This patch also simplifies by using
the built-in has_prefix/has_suffix functions and adds a first regression test
for static loop unwinding.

### CODE REVIEW

File: src/goto-programs/remove_static_init_loops.cpp [hunk 1]
```diff
@@ -65,23 +70,16 @@ void remove_static_init_loopst::unwind_enum_static(
...
-            options.set_option("unwindset", set+newset);
-          }
+          const std::string &set=options.get_option("unwindset");
+          std::string newset;
+          if(set!="")
+            newset=",";
+          newset+=
+            id2string(ins.function)+"."+std::to_string(loop_id)+":"+
+            std::to_string(unwinds);
+          options.set_option("unwindset", set+newset);
```
Comment (peterschrammel) [inline src/goto-programs/remove_static_init_loops.cpp]: Can you add a comment in https://github.com/diffblue/cbmc/blob/master/src/cbmc/cbmc_parse_options.cpp#L553 to make this side-effect more visible?

**Comment by tautschnig** [comment]
As a matter of traceability, and without precisely defined rules: may I suggest to either use --fixup <commit> or add some pseudo-header "Fixes: <commit>" when pull requests fix problems introduced in an earlier commit? (Just a suggestion, anyone feel free to say that this is just useless overhead.)

File: src/goto-programs/remove_static_init_loops.cpp (hunk 1 above)
Comment (mgudemann) [inline src/goto-programs/remove_static_init_loops.cpp]: added a comment

**Comment by mgudemann** [comment]
I added a `Fixes: sha_of_commit` to the commit message.

**Comment by kroening** [comment]
To determine whether a function is a Java function, I'd recommend you use the mode of the symbol, not it's prefix.

**Comment by mgudemann** [comment]
done using comparing the mode to `ID_java` as suggested

**Pull request merged**